[Feature guide] Deploy certificates from DigiCert and custom SCEP certificate authority on macOS (#27439)
fleetdm/fleet: articles/connect-end-user-to-wifi-with-certificate.md

# Connect end users to Wi-Fi or VPN with a certificate (DigiCert, NDES, or custom SCEP)

_Available in Fleet Premium_

Fleet can help your end users connect to Wi-Fi or VPN by deploying certificates from your certificate authority (CA). Fleet currently supports [DigiCert](https://www.digicert.com/digicert-one), [Microsoft NDES](https://learn.microsoft.com/en-us/windows-server/identity/ad-cs/network-device-enrollment-service-overview), and custom [SCEP](https://en.wikipedia.org/wiki/Simple_Certificate_Enrollment_Protocol) server.

## DigiCert

To connect end users to W-Fi or VPN with DigiCert certificates, we'll do the following steps:

- [Create service user in DigiCert](#step-1-create-service-user-in-digicert)
- [Create certificate profile in DigiCert](#step-2-create-certificate-profile-in-digicert)
- [Connect Fleet to DigiCert](#step-3-connect-fleet-to-digicert)
- [Add PKCS #12 configuration profile to Fleet](#step-4-add-pkcs-12-configuration-profile-to-fleet)

> ### Step 1: Create service user in DigiCert

Reviewer: @lukeheath @georgekarrv, @marko-lisica and I learned that we can't test this step because service users aren't available in our DigiCert demo instance. 

Do y'all think we (Fleet) can get a production DigiCert instance as a partner? This way, we can test the best practice workflow.

1. Head to [DigiCert One](https://one.digicert.com/)
2. Follow the instructions to create a service user [here](https://docs.digicert.com/en/platform-overview/manage-your-accounts/account-manager/users-and-access/service-users/create-a-service-user.html) and save the service user's API token.
> Make sure to assign **User and certificate manager** and **Certificate profile manager** roles
> when creating service user.

### Step 2: Create certificate profile in DigiCert

1. In DigiCert [Trust Lifcycle Manager](https://one.digicert.com/mpki/dashboard), select **Policies > Certificate profiles** from the main menu. Then select **Create profile from template** and select **Generic Device Certificate** from the list.
2. Add a friendly **Profile name** (e.g. "Fleet - Wi-Fi authentication").
3. Select your **Business unit** and **Issuing CA**.
4. Select **REST API** from **Enrollment method**. Then select **3rd party app** from the **Authentication method** dropdown and select **Next**.

> 5. Configure the certificate expiration as you wish.

Reviewer: ```suggestion
5. Configure the certificate expiration. At most organizations, this is set to 30 days.
```

> 5. Configure the certificate expiration. At most organizations, this is set to 90 days.

Reviewer: @marko-lisica we want the guide to be written for someone who's brand new to IT and someone who's worked for 10+ years in IT. Someone experienced (like our customers) can decided to follow them or not because they know.

@allenhouchins what do you think? 90 days sound right?

Reviewer: Also tagging @nonpunctual

Reviewer 2: @noahtalerman Yes! 90 days is a good middle ground for most organizations.

Author: Looks good to me! Thank you all!

> 6. In the **Subject DN and SAN fields** section, make sure to add **Common name** and **Other name (UPN)**. For **Common name**, select **REST request** from **Source for the field's value** dropdown and check **Required**. For **Other name (UPN)**, select **REST Request** and check both **Required** and **Multiple**.

Reviewer: > @noahtalerman: When does a user want to use the default values vs. variables? I think we want to add that in here.
> 
> @marko-lisica: I'm not sure. Not sure why is this UPN is used for.

@allenhouchins what do you think?

Reviewer: Also @nonpunctual, what do you think?

Reviewer 2: @noahtalerman Typically your networking or security team will suggest what to use for CN and/or UPN. CN is generally a higher level, domain identifier (ex: fleetdm.com) vs UPN is more specific, user or device level identifier and is used for authentication (ex: allen@fleetdm.com). In almost all cases, the UPN is going to use variables since its specific to the device or person the certificate is being issues to.

> 6. In the **Subject DN and SAN fields** section, make sure to add **Common name**. **Other name (UPN)** is optional. For **Common name**, select **REST request** from **Source for the field's value** dropdown and check **Required**. If you use **Other name (UPN)**, select **REST Request** and check both **Required** and **Multiple**. Organizations usually use device's serial number or user's email, you can use Fleet variables in the next section, and Fleet will replace these variables with the actual values before certificate is delivered to a device.

Author: @noahtalerman Wdyt?

> 6. In the **Subject DN and SAN fields** section, make sure to add **Common name**. **Other name (UPN)** is optional. For **Common name**, select **REST request** from **Source for the field's value** dropdown and check **Required**. If you use **Other name (UPN)**, select **REST Request** and check both **Required** and **Multiple**. Organizations usually use device's serial number or user's email, you can use Fleet variables in the next section, and Fleet will replace these variables with the actual values before certificate is delivered to a device.

Reviewer: ```suggestion
6. In the **Subject DN and SAN fields** section, make sure to add **Common name**. For **Common name**, select **REST request** from **Source for the field's value** dropdown and check **Required**. If you use **Other name (UPN)** (optional), select **REST Request** and check both **Required** and **Multiple**.
```

7. Click **Next** and leave all default options. We'll come back to this later.

### Step 3: Connect Fleet to DigiCert

1. In Fleet, head to **Settings > Integrations > Certificates**.
2. Select **Add CA** and then choose **DigiCert** in the dropdown.
3. Add a **Name** for your certificate authority. The best practice is to create a name based on your use case in all caps snake case (ex. "WIFI_AUTHENTICATION"). We'll use this name later as variable name in a configuration profile.
4. If you're using DigiCert One's cloud offering, keep the default **URL**. If you're using a self-hosted (on-prem) DigiCert One, update the URL to match the one you use to login to your DigiCert One.
5. In **API token**, paste your DigiCert server user's API token (from step 1).
6. In **Profile GUID**, paste your DigiCert One certificate profile GUID (from step 2). To get your profile GUID, in DigiCert, head to the [Certificate profiles](https://one.digicert.com/mpki/policies/profiles) page, open your profile, and copy **GUID**.

> 7. In **CN**, **UPN**, and **Certificate seat ID**, you can use fixed values or one of the [Fleet's host variables](https://fleetdm.com/docs/configuration/yaml-files#macos-settings-and-windows-settings).

Reviewer: ```suggestion
7. In **CN**, **UPN**, and **Certificate seat ID**, you can use fixed values or one of the [Fleet's host variables](https://fleetdm.com/docs/configuration/yaml-files#macos-settings-and-windows-settings). Organizations usually use the host's serial number or end user's email to deliver a certificate that's unique to the host.
```

6. Paste **Profile GUID** of certificate profile created in the **Step 2** section above. To get
   Profile GUID, go to [Certificate profiles](https://one.digicert.com/mpki/policies/profiles) page,
   open your profile, and copy **GUID**.
7. For **CN**, **UPN**, and **Certificate seat ID**, you can use fixed values or one of the [Fleet's
   host
   variables](https://fleetdm.com/docs/configuration/yaml-files#macos-settings-and-windows-settings).
8. Select **Add CA**, and your DigiCert certificate authority (CA) should appear in the list.

### Step 4: Add PKCS12 configuration profile to Fleet

[Add a cofniguration profile](https://fleetdm.com/guides/custom-os-settings) to Fleet, that includes the PKCS12 payload. In the profile, you will need to set `$FLEET_VAR_DIGICERT_PASSWORD_<CA_NAME>` as the `Password` and `$FLEET_VAR_DIGICERT_DATA_<CA_NAME>` as the `Data`.

Replace `<CA_NAME>` part of the variable, with name that you used in the section above, to connect
Fleet to DigiCert (e.g if name of the certificate authority is WIFI_AUTHENTICATION, variable name
will be `$FLEET_VAR_DIGICERT_PASSWORD_WIFI_AUTHENTICATION` and
`FLEET_VAR_DIGICERT_DATA_WIFI_AUTHENTICATION`).

> When sending the profile to hosts, Fleet will replace the variables variables with the proper values. Any errors will appear as a **Failed** status on the host details page, in **OS settings**.

Author: ```suggestion
When sending the profile to hosts, Fleet will replace the variables variables with the proper values. Any errors will appear as a **Failed** status on the host details page, in **OS settings**.

When the profile with the DigiCert certificate is resent on the host details page in **OS settings**, Fleet will get a new certificate from DigiCert and create a new seat, which will take 1 license. If you want to revoke a license used by a seat that was created when the initial certificate was issued, go to [Trust Lifcycle Manager > Account > Seats](https://demo.one.digicert.com/mpki/account/seats) and remove the respective seat.
```

> # Connect end users to Wi-Fi or VPN with a certificate (from DigiCert, NDES, or custom SCEP)

Reviewer: ```suggestion
# Connect end users to Wi-Fi or VPN with a certificate (DigiCert, NDES, or custom SCEP)
```

One less word

_Available in Fleet Premium_

Fleet can help your end users connect to Wi-Fi or VPN by adding your certificate authority to issue
certificates. Fleet currently supports
[DigiCert](https://www.digicert.com/digicert-one), [Microsoft
NDES](https://learn.microsoft.com/en-us/windows-server/identity/ad-cs/network-device-enrollment-service-overview),
and custom [SCEP](https://en.wikipedia.org/wiki/Simple_Certificate_Enrollment_Protocol) server.

> This guide will walk you through configuring your certificate authority and delivering certificates.

Reviewer: ```suggestion
```

I think we could cut this sentence for sake of less to maintain.

## DigiCert

> To install certificates from DigiCert to hosts, do the following steps:

Reviewer: ```suggestion
To install certificates from DigiCert to hosts, we'll do the following steps:
```

I think "we'll" makes this read a bit friendlier.

- [Create service user in DigiCert](#step-1-create-service-user-in-digicert)
- [Create certificate profile in DigiCert](#step-2-create-certificate-profile-in-digicert)
- [Connect Fleet to DigiCert](#step-3-connect-fleet-to-digicert)

> - [Add a PKCS12 configuration profile to Fleet](#step-4-add-pkcs-12-configuration-profile-to-fleet)

Reviewer: ```suggestion
- [Add configuration profile to Fleet](#step-4-add-pkcs-12-configuration-profile-to-fleet)
```

I think we can cut the `PKCS12` to make this a little bit easier to read/digest. Less words/concepts.

### Step 1: Create service user in DigiCert

1. Head to [DigiCert One](https://one.digicert.com/)

> 2. Follow instructions [here](https://docs.digicert.com/en/platform-overview/manage-your-accounts/account-manager/users-and-access/service-users/create-a-service-user.html), to create service user, and save service user API token.

Reviewer: ```suggestion
2. Follow the instructions o create a service user [here](https://docs.digicert.com/en/platform-overview/manage-your-accounts/account-manager/users-and-access/service-users/create-a-service-user.html) and save the service user's API token.
```

Adding articles.

> Make sure to assign **User and certificate manager** and **Certificate profile manager** roles
> when creating service user.

### Step 2: Create certificate profile in DigiCert

> 1. In DigiCert [Trust Lifcycle Manager](https://one.digicert.com/mpki/dashboard), select
>    **Policies > Certificate profiles** from the main menu, then select **Create profile from
>    template**, and select **Generic Device Certificate** from the list.

Reviewer: ```suggestion
1. In DigiCert [Trust Lifcycle Manager](https://one.digicert.com/mpki/dashboard), select **Policies > Certificate profiles** from the main menu. Then select **Create profile from template** and select **Generic Device Certificate** from the list.
```

2. Add a friendly **Profile name** (e.g. "Fleet - Wi-Fi authentication").
3. Select your **Business unit** and **Issuing CA**.

> 4. Select **REST API** from **Enrollment method**, then select **3rd party app** from
>    **Authentication method** dropdown, and select **Next**.

Reviewer: ```suggestion
4. Select **REST API** from **Enrollment method**. Then select **3rd party app** from the **Authentication method** dropdown and select **Next**.
```

> 5. Configure certificate expiration as you wish.

Reviewer: ```suggestion
5. Configure the certificate expiration as you wish.
```

Reviewer: @marko-lisica what should we recommend as best practice here?

Author: @noahtalerman I don't know, really. Probably a longer period of time, since another team might need to create these for IT.

Why not just leave that up to the customer?

> 6. In **Subject DN and SAN fields** section, make sure to add **Common name** and **Other name
>    (UPN)**. For **Common name**, select **REST request** from **Source for the field's value**
>    dropdown, and check **Required**. For **Other name (UPN)**, select **REST Request**, and check
>    both **Required** and **Multiple** checkboxes.

Reviewer: ```suggestion
6. In the **Subject DN and SAN fields** section, make sure to add **Common name** and **Other name (UPN)**. For **Common name**, select **REST request** from **Source for the field's value** dropdown and check **Required**. For **Other name (UPN)**, select **REST Request** and check both **Required** and **Multiple**.
```

> 7. You can click **Next** and leave all default options until you get to the last step, where you
>    need to select service user created in first step from **Select Service User** dropdown, and
>    select **Create**

Reviewer: ```suggestion
7. Click **Next** and leave all default options. We'll come back to this later.
```

When do we come back to this?

### Step 3: Connect Fleet to DigiCert

> 1. Go to Fleet, navigate to **Settings**, select **Integrations** tab, and select
>    **Certificates**.

Reviewer: ```suggestion
1. In Fleet, head to **Settings > Integrations > Certificates**.
```

We can shorten this up I think.

> 2. Select **Add CA** button, and select **DigiCert** from the dropdown on the top.

Reviewer: ```suggestion
2. Select **Add CA** and then choose **DigiCert** in the dropdown.
```

> 3. Add **Name** for your certificate authority. It's best to use all caps, beacuse it will be used
>    as variable name in configuration profile and name it based on your use case (e.g.
>    WIFI_AUTHENTICATION).

Reviewer: ```suggestion
3. Add a **Name** for your certificate authority. The best practice is to create a name based on your use case in all caps snake case (ex. "WIFI_AUTHENTICATION"). We'll use this name later as variable name in a configuration profile.
```

> 4. Keep default **URL**, or adjust if you're using on-prem DigiCert One. URL should match the one
>    you use to login to your DigiCert One account.

Reviewer: ```suggestion
4. If you're using DigiCert One's cloud offering, keep the default **URL**. If you're using a self-hosted (on-prem) DigiCert One, update the URL to match the one you use to login to your DigiCert One.
```

> 5. Paste **API token** from the **Step 1** section above.

Reviewer: ```suggestion
5. In **API token**, paste your DigiCert server user's API token (from step 1).
```

> 6. Paste **Profile GUID** of certificate profile created in the **Step 2** section above. To get
>    Profile GUID, go to [Certificate profiles](https://one.digicert.com/mpki/policies/profiles) page,
>    open your profile, and copy **GUID**.

Reviewer: ```suggestion
6. In **Profile GUID**, paste your DigiCert One certificate profile GUID (from step 2). To get your Profile GUID, in DigiCert, head to the [Certificate profiles](https://one.digicert.com/mpki/policies/profiles) page, open your profile, and copy **GUID**.
```

> 7. For **CN**, **UPN**, and **Certificate seat ID**, you can use fixed values or one of the [Fleet's
>    host
>    variables](https://fleetdm.com/docs/configuration/yaml-files#macos-settings-and-windows-settings).

Reviewer: ```suggestion
7. In **CN**, **UPN**, and **Certificate seat ID**, you can use fixed values or one of the [Fleet's host variables](https://fleetdm.com/docs/configuration/yaml-files#macos-settings-and-windows-settings).
```

Reviewer: When does a user want to use the default values vs. variables? I think we want to add that in here.

Author: @noahtalerman I'm not sure. Not sure why is this UPN is used for.

8. Select **Add CA**, and your DigiCert certificate authority (CA) should appear in the list.

### Step 4: Add PKCS12 configuration profile to Fleet

> [Add a configuration profile](https://fleetdm.com/guides/custom-os-settings) to Fleet, that includes the PKCS12 payload. In the profile, you will need to set `$FLEET_VAR_DIGICERT_PASSWORD_<CA_NAME>` as the `Password` and `$FLEET_VAR_DIGICERT_DATA_<CA_NAME>` as the `Data`.

Reviewer: ```suggestion
[Add a configuration profile](https://fleetdm.com/guides/custom-os-settings) to Fleet, that includes the PKCS12 payload. In the profile, for `Password` use `$FLEET_VAR_DIGICERT_PASSWORD_<CA_NAME>`. For `Data`, use `$FLEET_VAR_DIGICERT_DATA_<CA_NAME>`.
```

> Replace `<CA_NAME>` part of the variable, with name that you used in the section above, to connect
> Fleet to DigiCert (e.g if name of the certificate authority is WIFI_AUTHENTICATION, variable name
> will be `$FLEET_VAR_DIGICERT_PASSWORD_WIFI_AUTHENTICATION` and
> `FLEET_VAR_DIGICERT_DATA_WIFI_AUTHENTICATION`).

Reviewer: ```suggestion
Replace the `<CA_NAME>`, with name you created in step 3. For example, if the name of the CA is "WIFI_AUTHENTICATION" the variables will look like this: `$FLEET_VAR_DIGICERT_PASSWORD_WIFI_AUTHENTICATION` and `FLEET_VAR_DIGICERT_DATA_WIFI_AUTHENTICATION`.
```

> When sending the profile to hosts, Fleet will replace the variables variables with the proper values. Any errors will appear as a **Failed** status on the host details page, in **OS settings**.

Reviewer: ```suggestion
When Fleet delivers the profile to your hosts, Fleet will replace the variables. If something goes wrong, errors will appear on each host's **Host details > OS settings**.
```

> When the profile with the DigiCert certificate is resent on the host details page in **OS settings**, Fleet will get a new certificate from DigiCert and create a new seat, which will take 1 license. If you want to revoke a license used by a seat that was created when the initial certificate was issued, go to [Trust Lifcycle Manager > Account > Seats](https://demo.one.digicert.com/mpki/account/seats) and remove the respective seat.

Reviewer: ```suggestion
If you resend the profile (select **Resend** in **Host details > OS settings**), Fleet will get a new certificate and create a new seat in DigiCert, which will take 1 license. If you want to revoke a license, in DigiCert, head to [**Trust Lifcycle Manager > Account > Seats**](https://demo.one.digicert.com/mpki/account/seats) and remove the seat.
```

        <array>
            <dict>
                <key>Password</key>
                <string>$FLEET_VAR_DIGICERT_PASSWORD_CA_NAME</string>
                <key>PayloadContent</key>
                <data>$FLEET_VAR_DIGICERT_DATA_CA_NAME</data>
                <key>PayloadDisplayName</key>
                <string>CertificatePKCS12</string>
                <key>PayloadIdentifier</key>
                <string>com.fleetdm.pkcs12</string>
                <key>PayloadType</key>
                <string>com.apple.security.pkcs12</string>
                <key>PayloadUUID</key>
                <string>ee86cfcb-2409-42c2-9394-1f8113412e04</string>
                <key>PayloadVersion</key>
                <integer>1</integer>
            </dict>
        </array>
        <key>PayloadDisplayName</key>
        <string>DigiCert profile</string>
        <key>PayloadIdentifier</key>
        <string>TopPayloadIdentifier</string>
        <key>PayloadType</key>
        <string>Configuration</string>
        <key>PayloadUUID</key>
        <string>TopPayloadUUID</string>
        <key>PayloadVersion</key>
        <integer>1</integer>
    </dict>
</plist>
```

## Microsoft NDES

> To install certificates from Microsoft NDES to hosts, do the following steps:

Reviewer: ```suggestion
To connect end users to W-Fi or VPN with Microsoft NDES certificates, we'll do the following steps:
```

- [Connect Fleet to NDES](#step-1-connect-fleet-to-ndes)
- [Add SCEP configuration profile to Fleet](#step-2-add-scep-configuration-profile-to-fleet)

### Step 1: Connect Fleet to NDES

1. Go to the Fleet, navigate to **Settings**, select **Integrations** tab, and select **Certificates**.
2. Select **Add CA** button, and select **Microsoft NDES** from the dropdown on the top.
3. Add **SCEP URL** that accepts the SCEP protocol.
4. Add **Admin URL** and associated **Username** and **Password** to get the one-time challenge
   password for SCEP enrollment.
5. Select **Add CA**, and your NDES certificate authority (CA) should appear in the list.

> 
> 
> Note:
> * The example paths end with `/certsrv/mscep/mscep.dll` and `/certsrv/mscep_admin/` respectively. These path suffixes are the default paths for NDES on Windows Server 2022 and should only be changed if you have customized the paths on your server.
> * When saving the configuration, Fleet will attempt to connect to the SCEP server to verify the connection, including retrieving a one-time challenge password. This validation also occurs when adding a new SCEP configuration or updating an existing one via API and GitOps, including dry runs. Please ensure the NDES password cache size is large enough to accommodate this validation.

Reviewer: ```suggestion
The example paths end with `/certsrv/mscep/mscep.dll` and `/certsrv/mscep_admin/` respectively. These path suffixes are the default paths for NDES on Windows Server 2022 and should only be changed if you have customized the paths on your server.

When saving the configuration, Fleet will attempt to connect to the SCEP server to verify the connection, including retrieving a one-time challenge password. This validation also occurs when adding a new SCEP configuration or updating an existing one via API and GitOps, including dry runs. Please ensure the NDES password cache size is large enough to accommodate this validation.
```

### Step 2: Add SCEP configuration profile to Fleet

> [Add a configuration profile](https://fleetdm.com/guides/custom-os-settings) in Fleet that includes the SCEP payload. In the profile, you will need to set `$FLEET_VAR_NDES_SCEP_CHALLENGE` as the `Challenge` and `$FLEET_VAR_NDES_SCEP_PROXY_URL` as the `URL`.
> 
> Adjust the `Subject` values according to your organization's needs. You may set `$FLEET_VAR_HOST_END_USER_EMAIL_IDP` if the hosts were enrolled into Fleet MDM using an IdP (Identity Provider). You can also use any of the [Apple profile variables](https://support.apple.com/en-my/guide/deployment/dep04666af94/1/web/1.0) to uniquely identify your device.
> 
> When sending the profile to hosts, Fleet will replace the variables variables with the proper
> values. Any errors will appear as a **Failed** status on the host details page, in **OS settings**.

Reviewer: ```suggestion
1. Create a [configuration profile](https://fleetdm.com/guides/custom-os-settings) with the SCEP payload. In the profile, for `Challenge`, use`$FLEET_VAR_NDES_SCEP_CHALLENGE`. For `URL`, use `$FLEET_VAR_NDES_SCEP_PROXY_URL`.

2. If your Wi-Fi or VPN requires certificates that are unique to each host, update the `Subject`. You can use `$FLEET_VAR_HOST_END_USER_EMAIL_IDP` if your hosts automatically enrolled (via ADE) to Fleet with end user authentication enabled (learn more [here](https://fleetdm.com/docs/rest-api/rest-api#get-human-device-mapping)). You can also use any of the [Apple's built-in variables](https://support.apple.com/en-my/guide/deployment/dep04666af94/1/web/1.0).

When Fleet delivers the profile to your hosts, Fleet will replace the variables. If something goes wrong, errors will appear on each host's **Host details > OS settings**.
```

                            <string>OU</string>
                            <string>FLEET DEVICE MANAGEMENT</string>
                          </array>
                        </array>
                    </array>
             <key>URL</key>
             <string>$FLEET_VAR_NDES_SCEP_PROXY_URL</string>
          </dict>
          <key>PayloadDisplayName</key>
          <string>WIFI SCEP</string>
          <key>PayloadIdentifier</key>
          <string>com.apple.security.scep.9DCC35A5-72F9-42B7-9A98-7AD9A9CCA3AC</string>
          <key>PayloadType</key>
          <string>com.apple.security.scep</string>
          <key>PayloadUUID</key>
          <string>9DCC35A5-72F9-42B7-9A98-7AD9A9CCA3AC</string>
          <key>PayloadVersion</key>
          <integer>1</integer>
       </dict>
    </array>
    <key>PayloadDisplayName</key>
    <string>SCEP proxy cert</string>
    <key>PayloadIdentifier</key>
    <string>Fleet.WiFi</string>
    <key>PayloadType</key>
    <string>Configuration</string>
    <key>PayloadUUID</key>
    <string>4CD1BD65-1D2C-4E9E-9E18-9BCD400CDEDC</string>
    <key>PayloadVersion</key>
    <integer>1</integer>
</dict>
</plist>
```

## Custom SCEP server

> To install certificates from Microsoft NDES to hosts, do the following steps:

Reviewer 3: @marko-lisica is this a copy/pasta typo? should be `Custom SCEP` instead of `Microsoft NDES`

Author: Yes it's copy/pasta. Thanks for catching.

> To install certificates from Microsoft NDES to hosts, do the following steps:

Reviewer: ```suggestion
To connect end users to W-Fi or VPN with a custom SCEP server, we'll do the following steps:
```

- [Connect Fleet to custom SCEP server](#step-1-connect-fleet-to-custom-scep-server)
- [Add SCEP configuration profile to Fleet](#step-2-add-scep-configuration-profile-to-fleet2)

### Step 1: Connect Fleet to custom SCEP server

> 1. Go to the Fleet, navigate to **Settings**, select **Integrations** tab, and select **Certificates**.
> 2. Select **Add CA** button, and select **Custom** from the dropdown on the top.
> 3. Add **Name** for your certificate authority. It's best to use all caps, beacuse it will be used
>    as variable name in configuration profile and name it based on your use case (e.g.
>    WIFI_AUTHENTICATION).
> 4. Add **SCEP URL** that accepts the SCEP protocol.
> 5. Add **Challenge** password to authenticate Fleet with your SCEP server.
> 6. Select **Add CA**, and your custom SCEP certificate authority (CA) should appear in the list.

Reviewer: ```suggestion
1. In Fleet, head to **Settings > **Integrations > Certificates**.
2. Select the **Add CA** button and select **Custom** in the dropdown.
3. Add a **Name** for your certificate authority. The best practice is to create a name based on your use case in all caps snake case (ex. "WIFI_AUTHENTICATION"). We'll use this name later as variable name in a configuration profile.
4. Add your **SCEP URL** and **Challenge**.
6. Select **Add CA**.  Your custom SCEP certificate authority (CA) should appear in the list in Fleet.
```

                            <string>OU</string>
                            <string>FLEET DEVICE MANAGEMENT</string>
                          </array>
                        </array>
                    </array>
             <key>URL</key>
             <string>$FLEET_VAR_CUSTOM_SCEP_PROXY_URL_CA_NAME</string>
          </dict>
          <key>PayloadDisplayName</key>
          <string>WIFI SCEP</string>
          <key>PayloadIdentifier</key>
          <string>com.apple.security.scep.9DCC35A5-72F9-42B7-9A98-7AD9A9CCA3AC</string>
          <key>PayloadType</key>
          <string>com.apple.security.scep</string>
          <key>PayloadUUID</key>
          <string>9DCC35A5-72F9-42B7-9A98-7AD9A9CCA3AC</string>
          <key>PayloadVersion</key>
          <integer>1</integer>
       </dict>
    </array>
    <key>PayloadDisplayName</key>
    <string>SCEP proxy cert</string>
    <key>PayloadIdentifier</key>
    <string>Fleet.WiFi</string>
    <key>PayloadType</key>
    <string>Configuration</string>
    <key>PayloadUUID</key>
    <string>4CD1BD65-1D2C-4E9E-9E18-9BCD400CDEDC</string>
    <key>PayloadVersion</key>
    <integer>1</integer>
</dict>
</plist>
```

> ## How does Fleet SCEP proxy works (NDES and custom SCEP)

Reviewer: ```suggestion
## How the SCEP proxy works
```

We can shorten this I think.

> The SCEP proxy in Fleet acts as a middleman between the device and the NDES or custom SCEP server.
> When a device requests a certificate, the Fleet's SCEP proxy forwards the request to the NDES or
> custom SCEP server, retrieves the certificate, and sends it back to the device. In addition, the

Reviewer: ```suggestion
Fleet acts as a middleman between the host and the NDES or custom SCEP server. When a host requests a certificate from Fleet, Fleet requests a certificate from the NDES or
custom SCEP server, retrieves the certificate, and sends it back to the host. 

In addition, Fleet does the following:
```

> SCEP proxy:
> 
> - Retrieves the one-time challenge password from the NDES server.
>   The NDES admin password is encrypted in Fleet's database by the [server private key](https://fleetdm.com/docs/configuration/fleet-server-configuration#server-private-key). It cannot be retrieved via the API or the web interface.
>   Retrieving passwords for many devices may cause a bottleneck. To avoid long wait times, we recommend a gradual rollout of SCEP profiles.
>   - Restarting the NDES service will clear the password cache and may cause outstanding SCEP profiles to fail.
> - Resend the profile to the device if the one-time challenge password has expired.
>   - If the device has been offline and the one-time challenge password is more than 60 minutes old, the SCEP proxy assumes the password has expired and will resend the profile to the device with a new one-time challenge password.
> 
> The issued certificate will appear in the System Keychain on macOS. During the profile installation,
> the OS generates several temporary certificates needed for the SCEP protocol. These certificates may
> be briefly visible in the Keychain Access app on macOS. The CA certificate must also be installed
> and marked as trusted on the device for the issued certificate to appear as trusted. The IT admin can send the CA certificate in a separate [CertificateRoot profile](https://developer.apple.com/documentation/devicemanagement/certificateroot?language=objc)

Reviewer: ```suggestion
SCEP proxy:

- Retrieves the one-time challenge password from NDES. The NDES admin password is encrypted in Fleet's database by the [server private key](https://fleetdm.com/docs/configuration/fleet-server-configuration#server-private-key). It cannot be retrieved via the API or the web interface. Retrieving passwords for many hosts at once may cause a bottleneck. To avoid long wait times, we recommend a gradual rollout of SCEP profiles.
  - Restarting NDES will clear the password cache and may cause outstanding SCEP profiles to fail.
- Resends the configuration profile to the host if the one-time challenge password has expired.
  - If the host has been offline and the one-time challenge password is more than 60 minutes old, Fleet assumes the password has expired and will resend the profile to the host with a new one-time challenge password.

Certificates will appear in the System Keychain on macOS. During the profile installation,
the OS generates several temporary certificates needed for the SCEP protocol. These certificates may be briefly visible in the Keychain Access app on macOS. The CA certificate must also be installed and marked as trusted on the device for the issued certificate to appear as trusted. The IT admin can send the CA certificate in a separate [CertificateRoot profile](https://developer.apple.com/documentation/devicemanagement/certificateroot?language=objc)
```

> ### Assumptions and limitations

Reviewer: ```suggestion
## Assumptions and limitations
```

Make this an H2 so it appears in the right-side bar on the website?
